test/unit/org/apache/cassandra/index/sai/disk/SingleNodeQueryFailureTest.java

```diff
@@ -46,6 +47,10 @@ public class SingleNodeQueryFailureTest extends SAITester
     public void setup() throws Throwable
     {
         requireNetwork();
+
+        // Disable query optimization for this test so all indexes are queried,
+        // and we can inject failures into any of them.
+        QueryController.QUERY_OPT_LEVEL = 0;
```

Comment:
I am wondering, don't we need to restore the value in `tearDown`?

Reply:
Ok, double checked it but apparently it's not needed. The test framework resets the optimization anyways.